tests/MagicMethodAnnotationTest.php

```diff
@@ -809,15 +952,41 @@ function consumeInt(int $i): void {}
                     consumeInt(B::bar());',
             ],
             'magicStaticMethodInheritanceWithoutCallStatic' => [
-                'code' => '<?php
+                'code' => <<<'PHP'
+                    <?php
                     /**
                      * @method static int bar()
                      */
                     class A {}
                     class B extends A {}
-                    function consumeInt(int $i): void {}
 
-                    consumeInt(B::bar());',
+                    $a = B::bar();
+                    PHP,
+                'assertions' => [
+                    '$a===' => 'int',
+                ],
+                'ignored_issues' => [
+                    'UndefinedMethod',
+                ],
+            ],
+            'magicStaticMethodInheritanceWithoutCallStatic_AndWithManyArgs' => [
+                'code' => <<<'PHP'
+                    <?php
+                    /**
+                     * @method static int bar()
+                     */
+                    class A {}
+                    class B extends A {}
+
+                    $a = B::bar(123, "whatever");
+                    PHP,
+                'assertions' => [
+                    '$a===' => 'int',
+                ],
+                'ignored_issues' => [
+                    'UndefinedMethod',
+                    'TooManyArguments',
```

Comment:
I believe this is incorrect. If we claim the method does not exist, how do we know it has no parameters and returns int? Since `UndefinedMethod` is suppressed, it should accept any number of arguments and return mixed.

Reply:
In other words the test needs to be reverted to what I suggested and the code fixed to match the suggested test.